src/base/bittorrent/session.h

```diff
@@ -465,7 +467,7 @@ namespace BitTorrent
         bool isKnownTorrent(const InfoHash &hash) const;
         bool addTorrent(QString source, const AddTorrentParams &params = AddTorrentParams());
         bool addTorrent(const TorrentInfo &torrentInfo, const AddTorrentParams &params = AddTorrentParams());
-        bool deleteTorrent(const QString &hash, bool deleteLocalFiles = false);
+        bool deleteTorrent(const QString &hash, bool deleteLocalFiles = false, bool deleteForce = false);
```

Comment:
What is the use case behind `deleteForce`? can you provide a scenario?

Reply:
Scenario would be this:
- User downloads a torrent and has an external program run on completion (such as an extract).
- Extract places files in the torrent folder.
- Some of these files get moved to a permanent location, but some are left behind because they are garbage (things like scene nfo files as an example).
- Without "deleteForce" the torrent folder will not be removed by the deleteTorrent routine because the folder won't be empty after it delete's the torrent's files (the scene nfo in our example will still be there), thus leaving behind the torrent folder + garbage inside it.
- With "deleteForce" selected the garbage will also be deleted and the entire folder removed.
- This means that things like "cleanup after seeding is complete" will really and truly cleanup everything in the torrent download dir, instead of potentially leaving folders for torrents that no longer exist with their junk inside them (of course this may be opposite to the desired behavior, and so that's why it is an option in the SRL dialog, non-default)

Does that help clarify?

Reply:
>Does that help clarify?

I see, occasionally this could be useful.